### PR TITLE
issue/7124 fixing ArrayIndexOutOfBoundsException in media settings

### LINKED ISSUE #7124: Crash report 8.9 - 9.0.1: ArrayIndexOutOfBoundsException in MediaSettingsActivity.setupImageSizeSeekBar
```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.media.MediaSettingsActivity}: java.lang.ArrayIndexOutOfBoundsException: length=4; index=-1
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3319)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3415)
       at android.app.ActivityThread.access$1100(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1821)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.ArrayIndexOutOfBoundsException: length=4; index=-1
       at org.wordpress.android.ui.media.MediaSettingsActivity.setupImageSizeSeekBar(MediaSettingsActivity.java:645)
       at org.wordpress.android.ui.media.MediaSettingsActivity.showMetaData(MediaSettingsActivity.java:566)
       at org.wordpress.android.ui.media.MediaSettingsActivity.loadMedia(MediaSettingsActivity.java:345)
       at org.wordpress.android.ui.media.MediaSettingsActivity.loadMediaFromEditor(MediaSettingsActivity.java:317)
       at org.wordpress.android.ui.media.MediaSettingsActivity.onCreate(MediaSettingsActivity.java:246)
       at android.app.Activity.performCreate(Activity.java:6904)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1136)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3266)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3415)
       at android.app.ActivityThread.access$1100(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1821)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7406)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #7124 

Default to full size when we cant recognize image size parameter.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java [hunk 1]
```diff
@@ -651,6 +651,11 @@ private void setupImageSizeSeekBar() {
 
         int imageSizeKey = Arrays.asList(mImageSizeKeyArray).indexOf(mEditorImageMetaData.getSize());
 
+        //if the image size is something we can't recognize, default to the full size
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java]: It seems that there is at least one value that is expected but we don't test for: the empty string as [mentioned in the comment here](https://github.com/wordpress-mobile/WordPress-Android/blob/5fa249d23643993354eabf86c19a438ae2351787/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/MetadataUtils.java#L37-L37).

If that's indeed an expected value, what do you think about explicitly guard for it with an `if` statement? Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java [hunk 2]
```diff
@@ -651,6 +651,11 @@ private void setupImageSizeSeekBar() {
...
+        if (imageSizeKey == -1) {
+            imageSizeKey = mImageSizeLabelArray.length - 1;
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java]: I wonder what the unrecognized value may be. What do you think about adding some logging for it here? Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java (hunk 1 above)
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java]: Good catch! An empty string is indeed, not an expected value, the default value will always be `size-full`. All amend the comment.

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java (hunk 2 above)
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java]: Unrecognized value can really be anything - we are taking user input here. Having something that matches parameter regex (like `size-xl`) in the class attribute of an image will crash the media settings. 

Loggin is an excellent idea! Do you know if we have any helper class or tool for logging this kind of things (and retrieving them later?)?

Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java]: I see. It's not obvious that this is practically a user input field so, perhaps we can update the comment (Line 654) to include that bit of info.

About logging, I don't think we have a facility like that so, we can just use `AppLog` and hope to pick up the message via Helpshift when/if a user reports a problem.

**Comment by hypest** [comment]
LGTM!

**Pull request merged**